Dateisystem

GunJesus opened the pull request:
Dateissytem erfolgreich gekillt....

GunJesus committed:
Termin Struct

initial commit, Struct für den Termin erzeugt und getestet

GunJesus committed:
Termin Struct finished

Termin hält jetzt:
Titel
Beschreibung
Wiederkehrend
Datum
Endedatum

Time: Eingabe nur in form von: "2007-03-02T15:02:05 UTC" sonst fickt dich Go

GunJesus committed:
CreatTerminObj, speichert in JSON

Termin Objecte können mit einer einzelnen funktion erzeugt werden und zu jsons exportiert werden

CocoJasmin reviewed (approved):
Voll nice, danke. :) Nur kurz zu den Set Methoden von Termin, die kann man vlt noch Objekt spezifisch machen. Also dass nur ein Termin die Setter aufrufen kann.

GunJesus committed:
Dateisystem Done

Termine können in JSON gespeichert werden | "getTermin" liefert slices mit allen Terminen für einen User zurück | tests müssen nochmal überarbeitet werden

TheLoraxxz commented:
BItte Assert.True überall verwenden

TheLorenzor commented on dateisystem/Termin_Handler_test.go:
```diff
@@ -0,0 +1,61 @@
+	}
+
+	if termin.EndDate.String() != "2007-03-02 15:02:05 +0000 UTC" {
+		t.Errorf("End-Datum nicht hinterlegt")
```
Assert.True hier verwenden

TheLorenzor commented on dateisystem/Termin_Handler.go:
```diff
@@ -0,0 +1,87 @@
+	return t
+}
+
+func getTermine(username string) []Termin { //liefert slice mit allen terminen eines Users zurück
```
sollten die nicht public sein, dass man von außerhalb zugreifen kann?

TheLorenzor commented on dateisystem/Termin_Handler.go:
```diff
@@ -0,0 +1,87 @@
+		fmt.Println(err)
+	}
+	return t
+}
```
missing: export to ical and import from ical -.-> oder sollen wir das woanders rein packen?

GunJesus replied:
Export haben wir in eine seperate Komponente ausgelagtert. Die gehe ich als nächstes an

TheLorenzor reviewed (changes requested):
Bitte DTEsts straighten und fehlende Methoden hinzufügen?

GunJesus committed:
Dateisystem reworked

seter überarbeitet | alle relevanten Funktionen public gemacht | Tests sehen jetzt besser aus und verwenden assert | Funktion fürs Caching hinzugefügt

GunJesus committed:
DeleteAll | DeleteFromCache | StoreCache hinzugefügt

Die funktionen soltlen uns das leben etwas einfacher machen^^

GunJesus committed:
kleine Bug fixes und es hat sich noch ein ein ICS Parser eingeschlichen; Merge jetzt

GunJesus reviewed (commented):
macht was es soll, parser noch nicht umfangreich getestet

TheLorenzor reviewed (approved):
good wok
